pkg/lib/usage/count_collector.go

```diff
@@ -307,7 +312,57 @@ func (c *CountCollector) querySMSCount(appID string, rangeFrom *time.Time, range
 	}
 }
 
-func (c *CountCollector) querySMSSentEvents(appID string, rangeFrom *time.Time, rangeTo *time.Time, first uint64, after model.PageCursor) (events []*event.Event, lastCursor model.PageCursor, err error) {
+func (c *CountCollector) queryWhatsappCount(appID string, rangeFrom *time.Time, rangeTo *time.Time) (*whatsappCountResult, error) {
```

Comment:
Maybe we can use generics here to avoid duplicating the algorithm.

Reply:
I think we should not assume whatsapp count and sms count are having same regions, therefore duplicated the function to preserve the flexibility of having different regions in sms and whatsapp. I've reused the same code in `queryEvents` because I think they should always have the same logic.